### PR TITLE
[ENH] refactor annotation into segmentation and anomaly detection

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs
Dependent on #838 

First stage in #722 and #592. Also see #699

 Need to do this before v0.6

#### What does this implement/fix? Explain your changes.

creates two new modules, annotation and anomaly_detection. Only the first stage, makes the obvious changes, leaving some things in annotation.

Moves STRAY to anomaly_detection

Moves CLASP GGS and IGTS to segmentation.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
Ive broken this due to branch confusion, will park it for now until refactoring is sorted

File: aeon/datasets/_dataframe_loaders.py [hunk 1]
```diff
@@ -995,16 +994,4 @@ def load_tsf_to_dataframe(
...
-                and return_type not in MTYPE_LIST_HIERARCHICAL
-            ):
-                loaded_data = convert(
-                    loaded_data, from_type="pd-multiindex", to_type=return_type
-                )
-            else:
-                loaded_data = convert(
-                    loaded_data, from_type="pd_multiindex_hier", to_type=return_type
-                )
-
```
Comment (MatthewMiddlehurst) [inline aeon/datasets/_dataframe_loaders.py:1009]: think this is inherited from a previous PR

File: aeon/utils/plotting.py [hunk 2]
```diff
@@ -67,8 +66,6 @@ def plot_series(
         check_y(y, allow_index_names=True)
 
     series = list(series)
-    series = [convert_to(y, "pd.Series", "Series") for y in series]
```
Comment (MatthewMiddlehurst) [inline aeon/utils/plotting.py:70]: same as above for this file

File: docs/api_reference/annotation.rst [hunk 3]
```diff
@@ -4,19 +4,9 @@ Time series annotation
 ======================
 
 The :mod:`aeon.annotation` module contains algorithms and tools
-for time series annotation tasks, like anomaly/outlier detection,
-and time series segmentation.
+for time series annotation. This no longer includes anomaly/outlier
+detection and time series segmentation, which are now in their own module.
```
Comment (MatthewMiddlehurst) [inline docs/api_reference/annotation.rst:8]: what is contained within annotation then?

**Review by MatthewMiddlehurst — COMMENTED** [review]
Fine with this in principle, but have not had a thorough look at the full changes.

Not quite sure what happened with the changelog, and some datatype stuff seems to have snuck in. Unsure if this is from previous branches.

Focusing it on the two packages may make it a bit easier to read through. @lmmentel and @patrickzib have done work on these packages I believe, and may want to comment.

File: aeon/datasets/_dataframe_loaders.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/datasets/_dataframe_loaders.py]: reverted this to main

File: aeon/utils/plotting.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/utils/plotting.py:70]: reverted

File: docs/api_reference/annotation.rst (hunk 3 above)
Comment (TonyBagnall) [reply docs/api_reference/annotation.rst:8]: lots of messy code, see https://github.com/aeon-toolkit/aeon/issues/722 This is just the first stage, chose the easiest to move

**Review by hadifawaz1999 — APPROVED** [review]

**Review by MatthewMiddlehurst — COMMENTED** [review]
The new pages need to be included in https://aeon-toolkit--782.org.readthedocs.build/en/782/api_reference.html

Are the formatting changes in the changelog and pre-commit file necessary?

**Comment by MatthewMiddlehurst** [comment]
Ok well just as an FYI nothing links to the new .rst API pages currently

**Comment by TonyBagnall** [comment]
this is meant to be the first step in doing the refactor, its experimental, will link it all in to docs when its done, but its been stalled for ages. Any changes you want, feel free to PR

**Pull request merged**